layers/tests/e2e/layerPublisher.test.ts

```diff
@@ -0,0 +1,147 @@
+/**
+ * Test LayerPublisherStack class
+ *
+ * @group e2e/layers/all
+ */
+import { App, Stack } from 'aws-cdk-lib';
+import { Tracing } from 'aws-cdk-lib/aws-lambda';
+import { LayerPublisherStack } from '../../src/layer-publisher-stack';
+import {
+  deployStack,
+  destroyStack
+} from '../../../packages/commons/tests/utils/cdk-cli';
+import {
+  generateUniqueName,
+  invokeFunction,
+  isValidRuntimeKey,
+  createStackWithLambdaFunction
+} from '../../../packages/commons/tests/utils/e2eUtils';
+import {
+  RESOURCE_NAME_PREFIX,
+  SETUP_TIMEOUT,
+  TEARDOWN_TIMEOUT,
+  TEST_CASE_TIMEOUT
+} from './constants';
+import {
+  LEVEL,
+  InvocationLogs
+} from '../../../packages/commons/tests/utils/InvocationLogs';
+import { v4 } from 'uuid';
+import path from 'path';
+import packageJson from '../../package.json';
+
+const runtime: string = process.env.RUNTIME || 'nodejs1x';
+
+if (!isValidRuntimeKey(runtime)) {
+  throw new Error(`Invalid runtime key: ${runtime}`);
+}
+
+describe(`layers E2E tests (LayerPublisherStack) for runtime: ${runtime}`, () => {
+
+  const uuid = v4();
+  let invocationLogs: InvocationLogs[];
+  const stackNameLayers = generateUniqueName(RESOURCE_NAME_PREFIX, uuid, runtime, 'layerStack');
+  const stackNameFunction = generateUniqueName(RESOURCE_NAME_PREFIX, uuid, runtime, 'functionStack');
+  const functionName = generateUniqueName(RESOURCE_NAME_PREFIX, uuid, runtime, 'function');
+  const ssmParameterLayerName = generateUniqueName(RESOURCE_NAME_PREFIX, uuid, runtime, 'parameter');
+  const lambdaFunctionCodeFile = 'layerPublisher.class.test.functionCode.ts';
+
+  const invocationCount = 1;
+
+  const integTestApp = new App();
+  let stackLayer: LayerPublisherStack;
+  let stackFunction: Stack;
+
+  const powerToolsPackageVersion = packageJson.version;
+
+  beforeAll(async () => {
+
+    const layerName = generateUniqueName(RESOURCE_NAME_PREFIX, uuid, runtime, 'layer');
+
+    stackLayer = new LayerPublisherStack(
+      integTestApp,
+      stackNameLayers,
+      {
+        layerName: layerName,
+        powertoolsPackageVersion: powerToolsPackageVersion,
+        ssmParameterLayerArn: ssmParameterLayerName,
+      }
+    );
+
+    stackFunction = createStackWithLambdaFunction({
+      app: integTestApp,
+      stackName: stackNameFunction,
+      functionName: functionName,
+      functionEntry: path.join(__dirname, lambdaFunctionCodeFile),
+      tracing: Tracing.ACTIVE,
+      environment: {
+        UUID: uuid,
+        POWERTOOLS_PACKAGE_VERSION: powerToolsPackageVersion,
+        POWERTOOLS_SERVICE_NAME: 'LayerPublisherStack',
+      },
+      runtime: runtime,
+      bundling: {
+        externalModules: [
```

Comment:
I am not sure why we need to bundle the modules, we already have the layer, right?

Reply:
Counterintuitively enough, this [means](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_lambda_nodejs.BundlingOptions.html#externalmodules):
> A list of modules that should be considered as externals (already available in the runtime).

Given they are already in the layer like you said, we don't want `esbuild` to include them in the bundle.

Reply:
oh, it's the other way around 😅 

> A list of modules that should be considered as externals (already available in the runtime).